Documentation/backends.md

```diff
@@ -1,51 +1,106 @@
-
-  Note: Currently, AWS [limits](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Appendix_Limits.html) the number of entries per route table to 50.
-
-* gce: create IP routes in a [Google Compute Engine Network](https://cloud.google.com/compute/docs/networking#networks)
-  * Requirements:
-    * [Enable IP forwarding for the instances](https://cloud.google.com/compute/docs/networking#canipforward).
-    * [Instance service account](https://cloud.google.com/compute/docs/authentication#using) with read-write compute permissions.
-  * `Type` (string): `gce`
-
-  Command to create a compute instance with the correct permissions and IP forwarding enabled:
-  `$ gcloud compute instances create INSTANCE --can-ip-forward --scopes compute-rw`
-
-  Note: Currently, GCE [limits](https://cloud.google.com/compute/docs/resource-quotas) the number of routes for every *project* to 100.
-
-* alloc: only perform subnet allocation (no forwarding of data packets).
-  * `Type` (string): `alloc`
-
-* ali-vpc: create IP routes in a [alicloud VPC route table](https://vpc.console.aliyun.com)
-  * Requirements:
-    * Running on an ECS instance that is in an Alicloud VPC.
-    * Permission require accessid and keysecret
-  * `Type` (string): `ali-vpc`
-  * `AccessKeyID` (string): api access key id. can also be configure with environment ACCESS_KEY_ID
-  * `AccessKeySecret` (string): api access key secret.can also be configure with environment ACCESS_KEY_SECRET
-  Note: Currently, AliVPC limit the number of entries per route table to 50.
+# Backends
+
+Flannel may be paired with several different backends. Once set, the backend cannot be changed at runtime.
+
+VXLAN is the default choice. host-gw is recommended for more experienced users who want the performance improvement. UDP is suggested for debugging only.
+
+AWS, GCE, and AliVPC are experimental and unsupported. Proceed at your own risk.
+
+For more information on configuration options for cloud components, see:
+* [AliCloud VPC Backend for Flannel][alicloud-vpc]
+* [Amazon VPC Backend for Flannel][amazon-vpc]
+* [GCE Backend for Flannel][gce-backend]
+
+## Recommended backends
+
+### VXLAN
+
+Use in-kernel VXLAN to encapsulate the packets.
+
+Type and options:
+* `Type` (string): `vxlan`
+* `VNI` (number): VXLAN Identifier (VNI) to be used. Defaults to 1.
+* `Port` (number): UDP port to use for sending encapsulated packets. Defaults to kernel default, currently 8472.
+* `GBP` (Boolean): Enable [VXLAN Group Based Policy](https://github.com/torvalds/linux/commit/3511494ce2f3d3b77544c79b87511a4ddb61dc89).  Defaults to `false`.
+
+### host-gw
+
+Use host-gw to create IP routes to subnets via remote machine IPs. Requires direct layer2 connectivity between hosts running flannel.
+
+host-gw provides good performance, with few dependencies, and easy set up. http://machinezone.github.io/research/networking-solutions-for-kubernetes/
```

Comment:
I would probably not include this machinezone link as it doesn't give a very clear view of current vxlan performance